_posts/2019-08-09-iglu-scala-client-0.6.0-released/index.md

```diff
@@ -0,0 +1,116 @@
+
+This approach does not work for RT-first world anymore. There's no meaningful amount of attempts that resolver needs to make before considering a schema missing or invalid. Streaming application can keep working for many weeks without restart and if during this time, one registry goes down for couple of minutes and resolver will try to resolve a schema it means that until next TTL eviction all data will be invalid. And retries won't help here because they all will happen in a short period of time. 
+
+However, we still need to have certain retry behavior, because registries always can go offline. In a streaming world, the best practice for retries is backoff period. In 0.6.0 the Iglu resolver will attempt to refetch failed schemas with steadily growing period of time between attempts. This period grows from subsecond delays to approximately 20 minutes. What is also very important, these re-attempts will be made only for non-successful responses.
+
+ResolutionError (subtype of `ClientError`) has two properties to reflect the history of attempts: lastAttempt a timestamp of last attempt being made and attempts showing amount of attempts taken so far.
+
+
+<h3 id="new-validator">3. New Validator</h3>
+
+As it was mentioned before, Iglu Scala Client uses new [JSON Schema validator][networknt-validator] under the hood (which hover can be replaced with any custom one). Despite this validator also targets JSON Schema spec v4, it nevertheless can have incompatibilities with our previous JSON Schema validator. As a result some instances that were considered valid by Iglu Scala Client pre-0.6.0 can now be silently invalidated.
+
+Here's the short list of most widely used Snowplow components we're planning to release with Iglu Client 0.6.0:
+
+* Stream Enrich 0.22.0 (Snowplow R117)
+* RDB Shredder 0.15.0 (RDB Loader R31)
+* BigQuery Loader 0.2.0
+
+Please, monitor your bad rows produced by above assets.
+
+
+<h3 id="acknowledgements">4. Acknowledgements</h3>
+
+This is a huge release, overhauling the core part of Snowplow and we were developing and testing it since Fall 2018.
+During this time, we received an enormous amount of contributions from outside of core Snowplow Engineering team.
+Huge thanks to our Summer 2018 intern [Andrzej Sołtysik][andrzej], [Hacktoberfest][hacktoberfest] 2018 participant [Sajith Appukuttan][saj1th] and our partner from [The Globe and Mail Inc.][globe-and-mail] [Saeed Zareian][saeed].
+
+
+<h3 id="roadmap">5. Roadmap and upcomming features</h3>
+
+This release is planned to be a last one in 0.x series.
+Next release will likely include a relatively small amount of user-facing improvements and will have a 1.0.0 version, marking stability of API.
+Since 1.0.0 we're planning to introduce [MiMa-compatibility checks][mima] to our libraries in order to make update process more reliable.
```

Comment:
From 1.0.0 onwards we plan to introduce [MiMa-compatibility checks][mima] to our libraries in order to make the update process more reliable.